homeassistant/components/pi4ioe5v9xxxx/switch.py

```diff
@@ -0,0 +1,92 @@
+"""Allows to configure a switch using RPi GPIO."""
+import logging
+
+from pi4ioe5v9xxxx import pi4ioe5v9xxxx  # pylint: disable=import-error
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_PINS = "pins"
+CONF_INVERT_LOGIC = "invert_logic"
+CONF_I2CBUS = "i2c_bus"
+CONF_I2CADDR = "i2c_address"
+CONF_BITS = "bits"
+
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_BITS = 24
+DEFAULT_BUS = 1
+DEFAULT_ADDR = 0x20
+
+_SWITCHES_SCHEMA = vol.Schema({cv.positive_int: cv.string})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_PINS): _SWITCHES_SCHEMA,
+        vol.Optional(CONF_I2CBUS, default=DEFAULT_BUS): cv.positive_int,
+        vol.Optional(CONF_I2CADDR, default=DEFAULT_ADDR): cv.positive_int,
+        vol.Optional(CONF_BITS, default=DEFAULT_BITS): cv.positive_int,
+        vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the swiches devices."""
+    pins = config.get(CONF_PINS)
+    switches = []
+
+    pi4ioe5v9xxxx.setup(
+        i2c_bus=config[CONF_I2CBUS],
+        i2c_addr=config[CONF_I2CADDR],
+        bits=config[CONF_BITS],
+        read_mode=False,
+        invert=False,
+    )
+    for pin, name in pins.items():
+        switches.append(Pi4ioe5v9Switch(name, pin, config[CONF_INVERT_LOGIC]))
+    add_entities(switches)
+
+
+class Pi4ioe5v9Switch(ToggleEntity):
```

Comment:
This should inherit from `SwitchDevice`.
https://github.com/home-assistant/core/blob/078ce6766e17d197efcf03fe0182fb46836ddc3e/homeassistant/components/switch/__init__.py#L81

Reply:
Hi Martin, should I create a new pull request to inherit from SwitchDevice?

Reply:
I please. :+1: